### ISSUE TITLE
Any plan to have a english version?

### ISSUE THREAD

**Issue by ojss** [mangaki/mangaki#83]
Are there any plans to have a english version of the application? I am ready to help.

**Comment by RaitoBezarius** [comment]
@ojss: Definitely!

We would love to internationalize the project, and it's on our next big feature!
So to be honest, we're now trying to figure out when we should start to internationalize as we have a Django web app.

I was thinking of internationalizing once we would have a Django API + React.js web app. But @jilljenn do you think that we can go for i18n before?

**Comment by jilljenn** [comment]
Absolutely! We plan to translate the platform into English and Japanese first.
The first issue is there: https://github.com/mangaki/mangaki/issues/54

We plan to integrate Django REST Framework anyway, then we can start i18n, then merge Anime and Manga into a bigger class Work, then React.js ;)

**Comment by ojss** [comment]
:+1: Wow this is wonderful! I loved the concept of the application and I really want to contribute. I am currently learning python and django, I am primarily a Rails guy. I would really like to contribute some code when I can :)

How far have you reached on the REST framework? And may I ask why did you choose React?

**Comment by jilljenn** [comment]
Thank you for your interest :D How did you find us?
@RaitoBezarius is the biggest fan of React.js up here, so that's why. He started some work in the branch `upgrade/react`.

Right now, only a few parts are accessible from an API (like, mosaic or recommendations). We're right in the middle of a contest, but after this, we will be able to think deeper about the database ;) Meanwhile, feel free to try to install the whole thing, and don't hesitate to ask for help!

**Comment by ojss** [comment]
I found the project when I was searching for python/django based projects on github, to contribute to.
mangaki was in the top hits, and since I like manga it clicked instantly.
 
Thanks, I will clone the repo and look through it soon :)

**Comment by RaitoBezarius** [comment]
To answer for the React.js question, @ojss 

I think that an application like Mangaki have to be *strongly* interactive with the users, and in fact, this is achievable by doing a SPA application.

It gives a nice bonus for contributors and Mangaki because, we get separation of concerns, which means that backend contributors can concentrate themselves on backend development and frontend contributors can do the same for the frontend.

For the "Why React.js ?" part, I won't enter in an endless debate on whatever React.js is really necessary, or why not using Cycle.js or a tinier lib.

My point is that: React.js is *smaller* than Angular or Ember.js or any other SPA (excluding Web Components like Polymer).
Furthermore, where we could have used Polymer, there are *few people* over there which would be interested *to contribute* to a Polymer application, the learning curve is high and that would be unproductive for Mangaki.

React.js is definitely gaining more and *more traction* like a startup, and I feel that it can solve Mangaki a lot of problems, React have a big community, composed of great guys, we have better patterns and libraries every day (Redux, React-Router for example).

So in the end, the only choice in my opinion was React.js. :)
I welcome opinions about tech stack choices for Mangaki, as long as it is does not end up [like this](http://www.mongodb-is-web-scale.com/) :smile:

And I hope to use Mangaki as a reason to write better React.js and to use the by-products which could be generated from Mangaki use of React.js to improve the React ecosystem. It would be awesome.

---

On the another note, awesome, let me know if you have any problem with setup or questions about the current architecture! We're using a Vagrant VM to simplify the setup, if you feel like that you don't need it, you can always just use something like virtualenvwrapper and ensure you have PostgreSQL running with a DB/User for Mangaki, and let the recommendations flows! :rocket:

**Comment by ojss** [comment]
@RaitoBezarius Oh man! I read that and laughed so hard!  :laughing: 

The reason I asked you why React is because I was simply curious and I too am trying it out with a Rails app of mine(yeah I wanna learn it). I will start playing around with the code once I have some time away from college and exams :)

PS: A friend of mine already likes this. :smile:

**Comment by RaitoBezarius** [comment]
@ojss No problem! :smile: At least, it made you laugh and will ensure that people who ask the same question can get a careful answer to their doubts!

Play to your heart's content with React, it's definitely awesome :).

**Comment by jilljenn** [comment]
BTW @ojss, hope you're well, and that you can enjoy Mangaki in English now!
